### PR TITLE
refactor(adminPanel): RN-1284: Update admin panel tables PART 2

### PR DESCRIPTION
**Pull request by alexd-bes**
### Changes:
- Updated react router to v6 inline with the rest of the monorepo
- Updated route handling to be auto generated from objects instead of creating new pages
- Updated table styling 
- Updated tables to link to new pages instead of expanding rows

Note: changes to LESMIS coming in part 3

### CODE REVIEW

File: packages/admin-panel/src/profileRoutes.jsx [hunk 1]
```diff
@@ -11,19 +11,19 @@ import { ChangePasswordPage } from './pages/ChangePasswordPage';
 
 export const PROFILE_ROUTES = [
   {
-    to: '/profile',
-    tabs: [
+    url: '/profile',
+    childViews: [
```
Comment (jaskfla) [inline packages/admin-panel/src/profileRoutes.jsx:15]: These property names do make more sense 👌

File: packages/admin-panel/src/routes/entities/entities.js [hunk 2]
```diff
@@ -109,13 +92,20 @@ const IMPORT_CONFIG = {
...
+  title: 'Entities',
+  path: '',
+  default: true,
+  endpoint: ENTITIES_ENDPOINT,
+  columns: COLUMNS,
+  importConfig: IMPORT_CONFIG,
+  needsBESAdminAccess: ['delete'],
+  detailsView: {
+    title: 'Survey Responses',
+    endpoint: 'entities/{id}/surveyResponses',
```
Comment (jaskfla) [inline packages/admin-panel/src/routes/entities/entities.js:105]: `ENTITIES_ENDPOINT` is defined up top, though I’m not sure if this is actually an improvement?

```suggestion
    endpoint: `${ENTITIES_ENDPOINT}/{id}/surveyResponses`,
```

File: packages/admin-panel/src/routes/surveys/dataGroups.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  title: 'Data groups',
+  endpoint: 'dataGroups',
+  columns: [...DATA_GROUP_FIELDS, ...getDataSourceButtonsConfig(DATA_GROUP_FIELDS, 'dataGroup')],
+  path: '/data-groups',
+  createConfig: {
+    title: 'New Data Group',
+    actionConfig: {
+      title: 'Edit Data Group',
+      editEndpoint: 'dataGroups',
+      fields: [...DATA_GROUP_FIELDS],
```
Comment (jaskfla) [inline packages/admin-panel/src/routes/surveys/dataGroups.js:48]:
```suggestion
      fields: DATA_GROUP_FIELDS,
```

File: packages/admin-panel/src/routes/surveys/questions.js [hunk 4]
```diff
@@ -86,32 +84,31 @@ const OPTION_COLUMNS = [
...
+};
+
+export const questions = {
+  title: 'Questions',
+  path: '/questions',
+  endpoint: 'questions',
+  columns: QUESTION_COLUMNS,
+  isBESAdminOnly: true,
+  editorConfig: EDITOR_CONFIG,
+  getIsLink: ({ option_set_id: optionSetId }) => !!optionSetId,
```
Comment (jaskfla) [inline packages/admin-panel/src/routes/surveys/questions.js:106]: Admirable commitment to naming conventions 😆

File: packages/admin-panel/src/routes/users/users.jsx [hunk 5]
```diff
@@ -103,6 +101,7 @@ const COLUMNS = [
   {
     Header: 'Edit',
     type: 'edit',
+    source: 'id',
     width: 150,
```
Comment (jaskfla) [inline packages/admin-panel/src/routes/users/users.jsx:105]: I saw in another file this was renamed `colWidth`. Was that a universal change?

File: packages/admin-panel/src/table/DataFetchingTable/Cells.jsx [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+};
+
+const CellLink = styled(Link)`
+  color: inherit;
+  text-decoration: none;
+  max-width: 100%; // need this to make the ellipsis work
+`;
+
+const formatDetailUrl = (detailUrl, row) => {
+  console.log('formatDetailUrl', detailUrl, row);
```
Comment (jaskfla) [inline packages/admin-panel/src/table/DataFetchingTable/Cells.jsx:121]:
```suggestion
```

or

```suggestion
  console.debug('formatDetailUrl', detailUrl, row);
```

File: packages/admin-panel/src/table/DataFetchingTable/DataFetchingTable.jsx [hunk 7]
```diff
@@ -37,32 +34,38 @@ import {
...
-  .MuiTableSortLabel-icon {
-    opacity: 1;
+const TableContainer = styled(MuiTableContainer)`
+  position: relative;
+  flex: 1;
+  overflow: auto;
+  table {
+    min-width: 45rem;
+  }
+  thead {
```
Comment (jaskfla) [inline packages/admin-panel/src/table/DataFetchingTable/DataFetchingTable.jsx:52]:
```suggestion
  .MuiTableBody-root {
    min-width: 45rem;
  }
  .MuiTableCell-stickyHeader {
```

File: packages/admin-panel/src/table/DataFetchingTable/DataFetchingTable.jsx [hunk 8]
```diff
@@ -37,32 +34,38 @@ import {
...
+    position: sticky;
```
Comment (jaskfla) [inline packages/admin-panel/src/table/DataFetchingTable/DataFetchingTable.jsx:53]: Huh, does this not come with the `stickyHeader` prop? 🧐

File: packages/admin-panel/src/table/actions.js [hunk 9]
```diff
@@ -90,7 +90,7 @@ const refreshDataWithDebounce = debounce(
     const sortString = JSON.stringify(sortObjects);
 
     // Set up columns
-    const columnSources = [...columns.map(column => column.source).filter(source => source), 'id'];
+    const columnSources = columns.map(column => column.source).filter(source => source);
```
Comment (jaskfla) [inline packages/admin-panel/src/table/actions.js:93]: Admittedly a micro-optimisation that doesn’t matter with this small an array, but I would generally tend to filter before mapping—less stuff to map this way

Probably not worth changing in this particular instance

**Review by jaskfla — APPROVED** [review]
Might need a walkthrough to grasp the amended routing setup and resource page configuration, but it’s _feeling_ easier to follow which must count for something 😆🍪

A couple nitpicky comments, but I can’t remember what they were about right now (split this review over two days) 😅

File: packages/admin-panel/src/App.jsx [hunk 10]
```diff
@@ -1,86 +1,86 @@
...
+        <Route element={<AppPageLayout user={user} />}>
+          <Route index element={<Navigate to="/surveys" replace />} />
+          <Route path="*" element={<Navigate to="/surveys" replace />} />
+          {[...ROUTES, ...PROFILE_ROUTES].map(route => (
+            <Route
+              key={route.path}
+              path={route.path}
+              element={<TabPageLayout routes={route.childViews} baseUrl={route.path} />}
+            >
+              {/* <Route index element={<Navigate to={route.childViews[0].path} replace />} /> */}
```
[the remark this answers is not in the document]
Comment (alexd-bes) [reply packages/admin-panel/src/App.jsx:85]: Nope, I think I got rid of it in my part 3!

File: packages/admin-panel/src/App.jsx [hunk 11]
```diff
@@ -1,86 +1,86 @@
...
+    return [
+      ...acc,
+      {
+        ...childViewWithRoute,
+        detailsView: updatedDetailsView,
+      },
+      updatedDetailsView,
+    ];
+  }, []);
+};
```
[the remark this answers is not in the document]
Comment (alexd-bes) [reply packages/admin-panel/src/App.jsx:47]: Interesting, that's cool to know. I think in this case the array is always going to be small enough that performance won't make much of an impact, so I think it's okay to leave here

[the remark this answers is not in the document]
Comment (jaskfla) [reply packages/admin-panel/src/App.jsx:47]: Yep, I think so too—just thought I’d take this opportunity to bring it up

File: packages/admin-panel/src/table/DataFetchingTable/DataFetchingTable.jsx (hunk 8 above)
Comment (alexd-bes) [reply packages/admin-panel/src/table/DataFetchingTable/DataFetchingTable.jsx:53]: Only the `Table` component does, but in this case we need to add some extra stuff to make the whole `thead` sticky as we have 2 rows to be sticky now (header and filter row)

Comment (jaskfla) [reply packages/admin-panel/src/table/DataFetchingTable/DataFetchingTable.jsx:53]: Ah! In that case I think https://github.com/beyondessential/tupaia/pull/5610#discussion_r1590421881 should also be ignored

Thanks for clarifying 🙏

**Pull request merged**